### PR TITLE
PurgeOnStartup should throw because not supported

### PR DESCRIPTION
**Pull request by melkio**
During the Init phase of MessagePump there is a new check to verify PurgeOnStartup setting value. 

@Particular/azure-maintainers please review

Connect to #100

### CODE REVIEW

File: src/Transport/Seam/MessagePump.cs [hunk 1]
```diff
@@ -27,6 +28,11 @@ public Task Init(Func<PushContext, Task> pump, CriticalError criticalError, Push
             messagePump = pump;
             circuitBreaker = new RepeatedFailuresOverTimeCircuitBreaker("MessagePump", TimeSpan.FromSeconds(30), ex => criticalError.Raise("Failed to receive message from Azure Service Bus.", ex));
 
+            if (settings.PurgeOnStartup)
+            {
+                throw new ConfigurationErrorsException("Azure Service Bus transport doesn't support PurgeOnStartup behaviour");
```
Comment (SeanFeldman) [inline src/Transport/Seam/MessagePump.cs]: We shouldn't throw `ConfigurationErrorsException` for non config section related things. Maybe `InvalidOperationException`?

**Comment by SeanFeldman** [comment]
@melkio can't find a trace for the question you've asked - why not to drain messages when `PurgeOnStartup` is requested. Problem with that is that we could have messages that are not visible yet (deferred) and we can't receive them. Therefore purge won't actually get rid of all the messages.
@yvesgoeleven anything else that would be an issue with the draining approach?

**Comment by yvesgoeleven** [comment]
@SeanFeldman @melkio upgrades are typically done in a running state, which means that something can still be sending to those queues, you would be purging forever

**Comment by SeanFeldman** [comment]
:+1: for upgrade scenario

**Pull request merged**